pkg/controller/core/clusterqueue_controller.go

```diff
@@ -497,3 +552,62 @@ func (r *ClusterQueueReconciler) updateCqStatusIfChanged(
 	}
 	return nil
 }
+
+func (r *ClusterQueueReconciler) getWorkloadsStatus(cq *kueue.ClusterQueue) *kueue.ClusterQueuePendingWorkloadsStatus {
+	if r.shouldUpdatePendingWorkloadStatus(cq) {
+		return &kueue.ClusterQueuePendingWorkloadsStatus{
+			Head:           r.qManager.GetSnapshot(cq.Name),
+			LastChangeTime: metav1.Time{Time: time.Now()},
+		}
+	}
+	return cq.Status.PendingWorkloadsStatus
+}
+
+func (r *ClusterQueueReconciler) shouldUpdatePendingWorkloadStatus(cq *kueue.ClusterQueue) bool {
+	return cq.Status.PendingWorkloadsStatus == nil ||
```

Comment:
nit: This pattern: `a || b && c`  is readable IMO since the results depends on the order of checking the conditions. Please add brackets or be explicit 
```golang
if cq.Status.PendingWorkloadsStatus == nil {
 return true
 }
 return time.Since(cq.Status.PendingWorkloadsStatus.LastChangeTime.Time) >= r.queueVisibilityUpdateInterval && !equality.Semantic.DeepEqual(cq.Status.PendingWorkloadsStatus.Head, r.qManager.GetSnapshot(cq.Name))
 ```

Reply:
brackets were added https://github.com/kubernetes-sigs/kueue/pull/1069/commits/90b18bdee0de5861442bc95320133618fbf3ced0